dspace-rest/src/main/java/org/dspace/rest/BitstreamResource.java

```diff
@@ -691,26 +686,18 @@ public Response deleteBitstream(@PathParam("bitstream_id") Integer bitstreamId,
             writeStats(dspaceBitstream, UsageEvent.Action.UPDATE, user_ip, user_agent, xforwardedfor, headers,
                     request, context);
 
-            // Check if resource policy exists in bitstream.
-            boolean found = false;
-            List<org.dspace.authorize.ResourcePolicy> policies =  AuthorizeManager.getPolicies(context, dspaceBitstream);
-            for(org.dspace.authorize.ResourcePolicy policy : policies) {
-                if(policy.getID() == policyId) {
-                    found = true;
-                    break;
-                }
-            }
+            org.dspace.authorize.ResourcePolicy resourcePolicy = resourcePolicyService.find(context, policyId);
+            if(resourcePolicy.getdSpaceObject() == dspaceBitstream && authorizeService.authorizeActionBoolean(context, dspaceBitstream, org.dspace.core.Constants.REMOVE)) {
```

Comment:
This type of compare won't work as you are comparing objects & not primitive types.